dialog/form.go

```diff
@@ -0,0 +1,98 @@
+package dialog
+
+import (
+	"fyne.io/fyne"
+	"fyne.io/fyne/container"
+	"fyne.io/fyne/layout"
+	"fyne.io/fyne/theme"
+	"fyne.io/fyne/widget"
+)
+
+// FormDialog is a simple dialog window for displaying FormItems inside a form.
+type FormDialog struct {
+	*dialog
+	items   []*widget.FormItem
+	confirm *widget.Button
+	cancel  *widget.Button
+}
+
+// validateItems acts as a validation edge state handler that will respond to an individual widget's validation
+// state before checking all others to determine the net validation state. If the error passed is not nil, then the
+// confirm button will be disabled. If the error parameter is nil, then all other Validatable widgets in items are
+// checked as well to determine whether the confirm button should be disabled.
+// This method is passed to each Validatable widget's SetOnValidationChanged method in items by NewFormDialog.
+func (d *FormDialog) validateItems(err error) {
+	if err != nil {
+		d.confirm.Disable()
+		return
+	}
+	for _, item := range d.items {
+		if validatable, ok := item.Widget.(fyne.Validatable); ok {
+			if err := validatable.Validate(); err != nil {
+				d.confirm.Disable()
+				return
+			}
+		}
+	}
+	d.confirm.Enable()
+}
+
+// NewFormDialog creates and returns a dialog over the specified application using
+// the provided FormItems. The cancel button will have the dismiss text set and the confirm button will
+// use the confirm text. The response callback is called on user action after validation passes.
+// If any Validatable widget reports that validation has failed, then the confirm
+// button will be disabled. The initial state of the confirm button will reflect the initial
+// validation state of the items added to the form dialog.
+// The MinSize() of the CanvasObject passed will be used to set the size of the window.
```

Comment:
I don't think this line is needed as that is how all of Fyne layout works

Reply:
Removed